Android/app/src/main/java/io/github/project_travel_mate/mytrips/FriendsProfileActivity.java

```diff
@@ -203,6 +191,77 @@ public void onResponse(Call call, final Response response) throws IOException {
         });
     }
 
+    /**
+     * Fetches list of all trips user has travelled with given friend
+     */
+    private void getMutualTrips(String friendId) {
+
+        Handler handler = new Handler(Looper.getMainLooper());
+
+        String uri = API_LINK_V2 + "get-common-trips/" + friendId;
+        Log.v("EXECUTING", uri);
+
+        //Set up client
+        OkHttpClient client = new OkHttpClient();
+        //Execute request
+        final Request request = new Request.Builder()
+                .header("Authorization", "Token " + mToken)
+                .url(uri)
+                .build();
+        //Setup callback
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+
+                Log.e("Request Failed", "Message : " + e.getMessage());
+                handler.post(() -> networkError());
+            }
+
+            @Override
+            public void onResponse(Call call, final Response response) throws IOException {
+
+                handler.post(() -> {
+                    if (response.isSuccessful() && response.body() != null) {
+                        JSONArray arr;
+                        try {
+                            final String res = response.body().string();
+                            Log.v("Response", res);
+                            arr = new JSONArray(res);
+
+                            for (int i = 0; i < arr.length(); i++) {
+                                String id = arr.getJSONObject(i).getString("id");
+                                String start = arr.getJSONObject(i).getString("start_date_tx");
+                                String end = arr.getJSONObject(i).optString("end_date", null);
+                                String name = arr.getJSONObject(i).getJSONObject("city").getString("city_name");
+                                String tname = arr.getJSONObject(i).getString("trip_name");
+                                JSONArray array = arr.getJSONObject(i).getJSONObject("city").getJSONArray("images");
+                                String image = array.length() > 0 ? array.getString(0) : null;
+                                mTrips.add(new Trip(id, name, image, start, end, tname));
+                                // Specify a layout for RecyclerView
+                                // Create a horizontal RecyclerView
+                                mLayoutManager = new LinearLayoutManager(FriendsProfileActivity.this,
+                                        LinearLayoutManager.HORIZONTAL,
+                                        false
+                                );
+                                recyclerView.setLayoutManager(mLayoutManager);
+                                mMutualTripsAdapter = new MutualTripsAdapter(FriendsProfileActivity.this, mTrips);
```

Comment:
You are creating new adapter in the loop. A better option would be just create mTrips array in the loop & set adapter after the loop.

Reply:
Resolved.